### ISSUE TITLE
gensim.scripts.word2vec2tensor results in UnicodeDecodeError

### ISSUE THREAD

**Issue by sreedevigattu** [RaRe-Technologies/gensim#3043]
#### Problem description

- I created a word2vec model from the tokens read from 1.4L files using the following call
model.wv.save_word2vec_format(f"{folder}/wvmodel.wv", binary=True)
- Ran the following command to convert word-vectors from word2vec format into Tensorflow 2D tensor format
python -m gensim.scripts.word2vec2tensor -i model/wvmodel.wv -o model/ -b

The above command works for tokens read from 5000 files. But it fails when I read the tokens from 6000 files. Looks like there is some content in the one of the files (5000 to 6000) that the **word2vec2tensor** script has problems with.

Is there anyway I can fix this issue? Or atleast identify the offending file and remove it?

#### Steps/code/corpus to reproduce

Unfortunately I cannot share the dataset as it is huge.

2021-02-11 05:28:33,305 - utils_any2vec - INFO - loading projection weights from model/wvmodel.wv
**Traceback** (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.9/site-packages/gensim/scripts/word2vec2tensor.py", line 94, in <module>
    word2vec2tensor(args.input, args.output, args.binary)
  File "/usr/local/lib/python3.9/site-packages/gensim/scripts/word2vec2tensor.py", line 68, in word2vec2tensor
    model = gensim.models.KeyedVectors.load_word2vec_format(word2vec_model_path, binary=binary)
  File "/usr/local/lib/python3.9/site-packages/gensim/models/keyedvectors.py", line 1547, in load_word2vec_format
    return _load_word2vec_format(
  File "/usr/local/lib/python3.9/site-packages/gensim/models/utils_any2vec.py", line 285, in _load_word2vec_format
    _word2vec_read_binary(fin, result, counts,
  File "/usr/local/lib/python3.9/site-packages/gensim/models/utils_any2vec.py", line 204, in _word2vec_read_binary
    processed_words, chunk = _add_bytes_to_result(
  File "/usr/local/lib/python3.9/site-packages/gensim/models/utils_any2vec.py", line 186, in _add_bytes_to_result
    word = chunk[start:i_space].decode("utf-8", errors=unicode_errors)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbf in position 0: invalid start byte

#### Versions

Linux-4.14.214-160.339.amzn2.x86_64-x86_64-with-debian-10.6
Python 3.6.12 (default, Nov 18 2020, 14:46:32) 
[GCC 8.3.0]
Bits 64
NumPy 1.19.5
SciPy 1.5.4
gensim 3.8.3
FAST_VERSION 1

**Comment by piskvorky** [comment]
`load_word2vec_format()` accepts encoding parameters, `encoding='utf8'` and `unicode_errors='strict'`.

If your input is not in utf8, you'll need to override those parameters, e.g. `encoding='cp1252'`.

The code that calls `load_word2vec_format()` is in `word2vec2tensor.py`, so you may need to edit that. Or better, copy paste it to your own code – that module is trivial, it's effectively only 10 lines of code.

Also, I recommend you upgrade to Gensim 4.0.0: `pip install -U -pre gensim`

**Comment by sreedevigattu** [comment]
Thanks for the prompt response. But the **pip install -U -pre gensim** didn't work for me. Got the following error
Usage:   
  pip install [options] <requirement specifier> [package-index-options] ...
  pip install [options] -r <requirements file> [package-index-options] ...
  pip install [options] [-e] <vcs project url> ...
  pip install [options] [-e] <local project path> ...
  pip install [options] <archive url/path> ...
**no such option: -p**

I modified **load_word2vec_format**  to take unicode_errors='replace' but it failed with 
_AttributeError: 'Word2VecKeyedVectors' object has no attribute 'index_to_key'_

That must be because I couldn't install gensim 4.0.0. So I replaced **index_to_key** call to **index2word** and the .tsv files were generated.

**Comment by piskvorky** [comment]
It's `--pre`, not `-pre`. So all good now, can I close this ticket?

**Comment by sreedevigattu** [comment]
Oops! My bad. I ran with the --pre option and I could install gensim 4.0.0 and also reverted back to **index_to_key** . Thanks

When I ran it this time, I got this error.
python word2vec2tensor.py -i data/model/wvmodel.wv -o data/model/ -b
2021-02-12 14:19:54,594 - word2vec2tensor - INFO - running word2vec2tensor.py -i data/model/wvmodel.wv -o data/model/ -b
2021-02-12 14:19:54,594 - keyedvectors - INFO - loading projection weights from data/model/wvmodel.wv
2021-02-12 14:19:54,604 - keyedvectors - WARNING - duplicate word '' in word2vec file, ignoring all but first
2021-02-12 14:19:54,657 - keyedvectors - WARNING - duplicate word 'CLASS=BLOB-CODE' in word2vec file, ignoring all but first
2021-02-12 14:19:54,678 - keyedvectors - INFO - loaded (7818, 100) matrix from data/model/wvmodel.wv
dict_keys(['vector_size', 'index_to_key', 'next_index', 'key_to_index', 'vectors', 'norms', 'expandos', 'mapfile_path'])
Traceback (most recent call last):
  File "word2vec2tensor.py", line 83, in <module>
    word2vec2tensor(args.input, args.output, args.binary)
  File "word2vec2tensor.py", line 63, in word2vec2tensor
    file_metadata.write(gensim.utils.to_utf8(word) + gensim.utils.to_utf8('\n'))

I did a work-around and proceeded
for word in model.index_to_key:
            **if word == None:
                continue**
